### ISSUE TITLE
How to split a heterogeneous graph for link_pred targeting specific edge_type?

### ISSUE THREAD

**Issue by SohaK** [snap-stanford/deepsnap#37]
Hi 
I want to split a heterogeneous graph into train, dev, and test set for link_pred, but I need to have the positive and negative instances of the edges/links from a specific edge_type. Predicting that kind of edge is the problem description, other edges are just informative. Is there a way to do that with split? 

Thanks,
Soha

**Comment by anniekmyatt** [comment]
I might be wrong but I think deepsnap’s splitting functionality will give you positive and negative supervision edges for all edge types in a hetero graph. If you want to specify one edge type for your supervision edges maybe have a look at pytorch-geometric’s RandomLinkSplit, see for example how it’s used [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/load_csv.py). In the same folder there is also a hetero_link_pred example which I contributed, so I hope it doesn’t seem like showing off that I share it😅,  it just seems fairly relevant to what you’re trying to do).

**Comment by SohaK** [comment]
Thanks @anniekmyatt both RandomLinkSplit functionality and the code example are very useful for me.
